microsoft/testsuites/core/hv_module.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+from semver import VersionInfo
+
+from lisa import Logger, RemoteNode, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Redhat
+from lisa.sut_orchestrator.azure.tools import LisDriver
+from lisa.tools import Lsmod, Modinfo, Modprobe, Uname
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite covers test cases previously handled by LISAv2:
+    LIS-MODULES-CHECK, VERIFY-LIS-MODULES-VERSION,
+    INITRD-MODULES-CHECK, RELOAD-MODULES-SMP
+
+    It is responsible for ensuring the Hyper V drivers are all present,
+    are included in initrd, and are all the same version.
+    """,
+)
+class HvModule(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Verify the list of given LIS kernel modules and verify if the version
+           matches with the Linux kernel release number. (Drivers loaded directly in
+           to the kernel are skipped)
+        """,
+        priority=0,
+    )
+    def verify_lis_modules_version(
+        self, case_name: str, log: Logger, node: RemoteNode
+    ) -> None:
+        if not isinstance(node.os, Redhat):
+            log.info(f"{node.os.name} not supported.")
```

Comment:
This line can be merged into below exception message. If the test fails, the message will show in final result, it's helpful for troubleshooting. The log is not accessible easily as the final result.